shell/browser/api/electron_api_base_window.cc

```diff
@@ -737,8 +737,8 @@ void BaseWindow::RemoveMenu() {
 
 void BaseWindow::SetParentWindow(v8::Local<v8::Value> value,
                                  gin_helper::Arguments* args) {
-  if (IsModal()) {
-    args->ThrowError("Can not be called for modal window");
+  if (IsModal() && !(value->IsNull() || value->IsUndefined())) {
+    args->ThrowError("Modal window can be only detached from the parent.");
```

Comment:
I'm not sure if this would be clear to a user 🤔 

I believe you're looking to indicate that modal windows attachment is unidirectional: aka that a modal window can be detached from a parent and then once that happens can never again be reattached. If that's the case, it should probably be more clear.